File: EDoctorsApp/EDoctors/app/src/main/java/com/incubation_lab/edoctors/Splash/ui/ViewPagerAdapter.java
package com.incubation_lab.edoctors.Splash.ui;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class ViewPagerAdapter  extends FragmentStateAdapter {


    public ViewPagerAdapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        if(position == 0){
            return SplashToUserLoginFragment.newInstance();
        }
        return SplashToDoctorLoginFragment.newInstance();
    }

    @Override
    public int getItemCount() {
        return 2;
    }
}

File: EDoctorsApp/app/src/main/java/com/incubation_lab/edoctor/ViewPagerAdapter.kt
package com.incubation_lab.edoctor

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter

class ViewPagerAdapter(fragmentManager: FragmentManager, lifecycle: Lifecycle) : FragmentStateAdapter(fragmentManager, lifecycle) {
    override fun getItemCount(): Int {
        return 2
    }

    override fun createFragment(position: Int): Fragment {
        if(position ==0){
            return UserLoginFragment.newInstance()
        }
        else{
            return DoctorLoginFragment.newInstance()
        }
    }

}
